karoo_gp/test/test_base.py

```diff
@@ -0,0 +1,55 @@
+import hashlib
+
+import pytest
+
+from karoo_gp import Base_GP
+
+@pytest.fixture(scope="module")
+def kwargs():
+    return dict(
+        kernel='m',
+        tree_type='r',
+        tree_depth_base=3,
+        tree_depth_max=3,
+        tree_depth_min=1,
+        tree_pop_max=100,
+        gen_max=10,
+        tourn_size=7,
+        filename='',
+        save_dir='test',
+        evolve_repro=0.1,
+        evolve_point=0.1,
+        evolve_branch=0.2,
+        evolve_cross=0.6, 
+        display='s',
+        precision=6,
+        swim='p',
+        mode='s'
+    )
+
+@pytest.fixture(scope="module")
```

Comment:
I think it would be better to start with a new object for each test.
```suggestion
@pytest.fixture
```